tpgtools/templates/provider_dcl_endpoints.go.tmpl

```diff
@@ -0,0 +1,65 @@
+{{/* Copyright 2021 Google LLC. All Rights Reserved.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+			http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License. */}}
+	// ----------------------------------------------------------------------------
+	//
+	//     ***     AUTO GENERATED CODE    ***    AUTO GENERATED CODE     ***
+	//
+	// ----------------------------------------------------------------------------
+	//
+	//     This file is automatically generated by Magic Modules and manual
+	//     changes will be clobbered when the file is regenerated.
+	//
+	//     Please read more about how to change this file in
+	//     .github/CONTRIBUTING.md.
+	//
+	// ----------------------------------------------------------------------------
+
+package google
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+{{range $index, $pkg := .}}
+var {{$pkg.ProductType}}EntryKey = "eventarc_custom_endpoint"
+var {{$pkg.ProductType}}Entry = &schema.Schema{
+	Type:         schema.TypeString,
+	Optional:     true,
+	ValidateFunc: validateCustomEndpoint,
+	DefaultFunc: schema.MultiEnvDefaultFunc([]string{
+		"GOOGLE_EVENTARC_CUSTOM_ENDPOINT",
+	}, ""),
+}
+{{end}}
+
+//Add new values to config.go.erb config object delcaration
```

Comment:
Should we look into writing these values in a machine-readable format so they can be read & used by MMv1?

I'm thinking that writing out the product metadata to a JSON file in a well-known place can let us add this information to the generated files like config.go

Reply:
I talked to Riley about this he said let's not bundle writing config.go and provider.go with this change.

Reply:
I don't like the json idea because it has to read a json file at runtime which sounds slow?

Reply:
Ah, I mean if we write a machine-readable format during tpgtools compilation we can read it during MMv1 compilation. It would all be contained in this repo and nothing should make it downstream to TPG/TPGB

Reply:
It would probably require swapping the order of tpgtools/mmv1 compilation, but there's no reason those need to be in a particular order today

Reply:
ahh I see whaat you mean, hmm that means tpg is bottlenecked by mmv1. In order to build tpgtools you always have to build mmv1

Reply:
That's already mostly true. Plenty of handwritten files live in MMv1 as well as config.go/provider.go. It would only need to be recompiled when a new product or resource is added, or a name is changed